synce_manager.c

```diff
@@ -0,0 +1,289 @@
+/**
+ * @file synce_manager.c
+ * @brief Interface for managing QL for external clock sources
+ * @note SPDX-FileCopyrightText: Copyright 2022 Intel Corporation
+ * @note SPDX-License-Identifier: GPL-2.0+
+ */
+#include <unistd.h>
+#include <pthread.h>
+#include <stdlib.h>
+#include <net/if.h>
+#include <ctype.h>
+#include "print.h"
+#include "config.h"
+#include "synce_dev.h"
+#include "synce_clock.h"
+#include "synce_manager.h"
+
+struct synce_clock;
+
+enum synce_manager_type {
+	MSG_DEV_NAME = 1,
+	MSG_SRC_NAME,
+	MSG_ERR_MSG,
+	MSG_GET_QL,
+	MSG_GET_EXT_QL,
+	MSG_SET_QL,
+	MSG_SET_EXT_QL,
+};
+
+struct synce_manager_tlv {
+	uint16_t type;	// enum synce_manager_type
+	uint16_t length;// length in bytes
+	void *value;	// data of given length
+};
+
+void synce_manager_parse_input(const char *input,
```

Comment:
I think we are still having some kind of misunderstanding of this TLV's approach, the command from the userspace in the TLV approach is not a string, it shall be a byte stream, and as such shall be handled differently, not by reading/parsing a string. What is the difference of parsing a string and parsing a fake-TLV encoded in a string? Please let me know if if you still want the TLV so i will elaborate, but you could just fall back to the string parsing as you did in previous version.

Reply:
changed it to bytestream.